x-pack/plugins/uptime/server/lib/synthetics_service/service_api_client.ts

```diff
@@ -85,6 +85,38 @@ export class ServiceAPIClient {
     return this.callAPI('POST', { ...data, runOnce: true });
   }
 
+  async checkIfAccountAllowed() {
+    if (this.authorization) {
+      // in case username/password is provided, we assume it's always allowed
+      return true;
+    }
+
+    const httpsAgent = this.getHttpsAgent();
+
+    if (this.locations.length > 0 && httpsAgent) {
+      // get a url from a random location
+      const url = this.locations[Math.floor(Math.random() * this.locations.length)].url;
```

Comment:
I assume we should eventually have a gateway to load balance between available locations? Maybe worth discussing with the cloud team.

Not for now though, we can keep it simple, just saying in case we want to add that to the sync agenda later.